http/middleware.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Middleware function based on {@linkcode Deno.ServeHandlerInfo} but an added
+ * `next()` function which calls the next middleware function in the middleware
+ * chain.
+ */
+export type MiddlewareFunction = (
+  request: Request,
+  info: Deno.ServeHandlerInfo,
+  next: () => Response | Promise<Response>,
+) => Response | Promise<Response>;
+
+/**
+ * Creates a {@linkcode Deno.ServeHandler} from the given middleware chain,
+ * which can then be passed to {@linkcode Deno.serve}.
+ *
+ * @param middlewares Middleware chain
+ *
+ * @example
+ * ```ts
+ * import {
+ *   type MiddlewareFunction,
+ *   createHandler,
+ * } from "https://deno.land/std@$STD_VERSION/http/middleware.ts";
+ *
+ * const middleware1: MiddlewareFunction = async (_request, _info, next) => {
+ *   const start = performance.now();
+ *   const response = await next();
+ *   const duration = performance.now() - start;
+ *   response.headers.set("x-request-time", duration.toString());
+ *   return response;
+ * };
+ *
+ * const middleware2: MiddlewareFunction = (request, info) => {
+ *   return Response.json({ request, info });
+ * };
+ *
+ * const handler = createHandler([])
```

Comment:
Do you mean the below?
```suggestion
 * const handler = createHandler([middleware1, middleware2])
```